src/test/e2e/psp.feature.ts

```diff
@@ -0,0 +1,189 @@
+    dapiServer = await dapiServerFactory.deploy(
+      accessControlRegistry.address,
+      dapiServerAdminRoleDescription,
+      roles.manager.address,
+      airnodeProtocol.address
+    );
+
+    // Access control
+    const managerRootRole = await accessControlRegistry.deriveRootRole(roles.manager.address);
+    await accessControlRegistry
+      .connect(roles.manager)
+      .initializeRoleAndGrantToSender(managerRootRole, dapiServerAdminRoleDescription);
+
+    // Wallets
+    const airnodeWallet = ethers.Wallet.fromMnemonic(localConfig.airnodeMnemonic);
+    const airnodePspSponsorWallet = node.evm
+      .deriveSponsorWalletFromMnemonic(localConfig.airnodeMnemonic, roles.sponsor.address, PROTOCOL_ID_PSP)
+      .connect(provider);
+    await roles.deployer.sendTransaction({
+      to: airnodePspSponsorWallet.address,
+      value: ethers.utils.parseEther('1'),
+    });
+
+    // Templates
+    const endpointId = ethers.utils.keccak256(
+      ethers.utils.defaultAbiCoder.encode(
+        ['string', 'string'],
+        [localConfig.endpoint.oisTitle, localConfig.endpoint.endpointName]
+      )
+    );
+    const parameters = abi.encode(localConfig.templateParameters);
+    const templateId = ethers.utils.solidityKeccak256(['bytes32', 'bytes'], [endpointId, parameters]);
+
+    // Subscriptions
+    const threshold = (await dapiServer.HUNDRED_PERCENT()).div(localConfig.threshold); // Update threshold %
+    const beaconUpdateSubscriptionConditionParameters = ethers.utils.defaultAbiCoder.encode(['uint256'], [threshold]);
+    const beaconUpdateSubscriptionConditions = [
+      {
+        type: 'bytes32',
+        name: '_conditionFunctionId',
+        value: ethers.utils.defaultAbiCoder.encode(
+          ['bytes4'],
+          [dapiServer.interface.getSighash('conditionPspBeaconUpdate')]
+        ),
+      },
+      { type: 'bytes', name: '_conditionParameters', value: beaconUpdateSubscriptionConditionParameters },
+    ];
+    const encodedBeaconUpdateSubscriptionConditions = abi.encode(beaconUpdateSubscriptionConditions);
+    await dapiServer
+      .connect(roles.randomPerson)
+      .registerBeaconUpdateSubscription(
+        airnodeWallet.address,
+        templateId,
+        encodedBeaconUpdateSubscriptionConditions,
+        airnodeWallet.address,
+        roles.sponsor.address
+      );
+  });
+
+  it('updates the beacon successfully', async () => {
+    jest
+      .spyOn(config, 'loadAirnodeConfig')
+      .mockImplementationOnce(() => airnodeConfig as any)
+      .mockImplementationOnce(() => airnodeConfig as any);
+    jest.spyOn(config, 'loadAirkeeperConfig').mockImplementationOnce(() => airkeeperConfig as any);
+    const res = await psp.handler();
+
+    expect(dapiServer).toBeDefined();
+    expect(res).toEqual({
+      statusCode: 200,
+      body: JSON.stringify({ ok: true, data: { message: 'PSP beacon update execution has finished' } }),
+    });
```

Comment:
I'm not sure that this is enough to determine if the beacon was successfully updated. I think we should be spying on the DapiServer functions in order to check that the expected ones where actually called with the expected args. WDYT?

Reply:
Can we instead of spying just read the value and see if it's updated? (just asking, not sure if it would work)

Reply:
Yes agreed, great suggestion, I'll add :+1: